test/integ-new/main_new_test.go

```diff
@@ -0,0 +1,212 @@
+					Iss: "jvs.abcxyz.dev",
+					Req: cfg.ServiceAccount,
+					Sub: cfg.ServiceAccount,
+				},
+			},
+		},
+		{
+			name:          "breakglass",
+			justification: "issues/12345",
+			isBreakglass:  true,
+			wantTestValidationResult: &testValidationResult{
+				Breakglass: true,
+				Justifications: []v0.Justification{
+					{
+						Category: "breakglass",
+						Value:    "issues/12345",
+					},
+				},
+				Claims: testClaim{
+					Aud: []string{"dev.abcxyz.jvs"},
+					Exp: ts.Add(testTTLTime).UTC().Format(timestampFormat),
+					Iat: ts.Format(timestampFormat),
+					Iss: "jvsctl",
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			var createCmd cli.TokenCreateCommand
+			_, stdout, _ := createCmd.Pipe()
+
+			createTokenArgs := []string{"-server", cfg.APIServer, "-justification", tc.justification, "-ttl", testTTLString, "--auth-token", cfg.IDToken}
+			if tc.isBreakglass {
+				createTokenArgs = append(createTokenArgs, "-breakglass")
+			}
+
+			if err := createCmd.Run(ctx, createTokenArgs); err != nil {
+				t.Fatal(err)
+			}
+
+			token := stdout.String()
+
+			validateTokenArgs := []string{"-token", token, "-jwks-endpoint", cfg.JWTEndpoint, "--format", "json"}
+
+			var validateCmd cli.TokenValidateCommand
+			_, stdout, _ = validateCmd.Pipe()
+
+			if err := validateCmd.Run(ctx, validateTokenArgs); err != nil {
+				t.Fatal(err)
+			}
+
+			gotTestValidationResult := &testValidationResult{}
+			if err := json.Unmarshal(stdout.Bytes(), gotTestValidationResult); err != nil {
+				t.Fatalf("failed to unmarshal validation result: %v", err)
+			}
+
+			cmpOpts := []cmp.Option{
+				protocmp.Transform(),
+				cmpopts.IgnoreFields(testClaim{}, "Exp", "Iat", "Iss"),
+			}
+
+			if diff := cmp.Diff(tc.wantTestValidationResult, gotTestValidationResult, cmpOpts...); diff != "" {
+				t.Errorf("wantValidationResult got unexpacted diff (-want, +got):\n%s", diff)
+			}
+
+			if err := testDiffClaimTimestamp(t, tc.wantTestValidationResult.Claims, gotTestValidationResult.Claims); err != nil {
+				t.Errorf("Claims got unexpected diff: %v", err)
+			}
+		})
+	}
+}
+
+// testDiffClaim compares the timestamp within the claims.
+// The seconds in timestamp will be trimed before comparing,
+// this is do help mitigate the delay when running tests.
+func testDiffClaimTimestamp(tb testing.TB, want, got testClaim) error {
+	tb.Helper()
+
+	var rErr error
+
+	if want.Exp[:len(want.Exp)-3] != got.Exp[:len(got.Exp)-3] {
+		rErr = errors.Join(rErr, fmt.Errorf(fmt.Sprintf("got unexpected exp timestamp (-want, +got)\n -%s\n +%s\n", want.Exp, got.Exp)))
+	}
+
+	if want.Exp[:len(want.Iat)-3] != got.Exp[:len(got.Iat)-3] {
+		rErr = errors.Join(rErr, fmt.Errorf(fmt.Sprintf("got unexpected iat timestamp (-want, +got)\n -%s\n +%s\n", want.Iat, got.Iat)))
+	}
+
+	if want.Exp[:len(want.Iss)-3] != got.Exp[:len(got.Iss)-3] {
```

Comment:
Is this a typo? Iss is issuer instead of timestamp. We probably should not ignore compare this field as well.

Reply:
Fixed.